dynmsg/src/msg_parser_c.cpp

```diff
@@ -480,6 +492,9 @@ void yaml_to_rosmsg_impl(
         throw std::runtime_error("unknown type");
     }
   }
+  if (!yaml_keys.empty()) {
+    throw std::runtime_error("Found unknown fields in the YAML not corresponding to the given message.");
```

Comment:
Is it possibly worth adding some number of keys to the error? There is no upper limit I suppose, so maybe not... Though even if was a single key it might help with debugging if users knew exactly where the problem was.

Reply:
Good idea, added all the keys, since a message doesn't typically have a lot of top-level keys. Also added the message name to the error message.